manifests/supervisorcluster/1.22/cns-csi.yaml

```diff
@@ -390,3 +390,208 @@ spec:
+---
+apiVersion: cert-manager.io/v1alpha3
+kind: Issuer
+metadata:
+  labels:
+    app: vsphere-csi-webhook
+  name: vmware-system-csi-selfsigned-issuer
+  namespace: vmware-system-csi
+spec:
+  selfSigned: {}
+---
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingWebhookConfiguration
+metadata:
+  name: vmware-system-csi-validating-webhook-configuration
+  labels:
+    app: vsphere-csi-webhook
+  annotations:
+    cert-manager.io/inject-ca-from: vmware-system-csi/vmware-system-csi-serving-cert
+webhooks:
+  - name: validation.csi.vsphere.vmware.com
+    clientConfig:
+      service:
+        name: vmware-system-csi-webhook-service
+        namespace: vmware-system-csi
+        path: "/validate"
+    rules:
+      - apiGroups:   [""]
+        apiVersions: ["v1", "v1beta1"]
+        operations:  ["CREATE", "UPDATE"]
+        resources:   ["persistentvolumeclaims"]
+        scope: "Namespaced"
+    sideEffects: None
+    admissionReviewVersions: ["v1"]
+    failurePolicy: Fail
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: vsphere-csi-webhook-cluster-role
+rules:
+  - apiGroups: [""]
+    resources: ["persistentvolumes", "persistentvolumeclaims"]
+    verbs: ["get", "list", "watch"]
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: vsphere-csi-webhook-cluster-role-binding
+subjects:
+  - kind: ServiceAccount
+    name: default
+    namespace: vmware-system-csi
+roleRef:
+  kind: ClusterRole
+  name: vsphere-csi-webhook-cluster-role
+  apiGroup: rbac.authorization.k8s.io
+---
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: vsphere-csi-webhook-role
+  namespace: vmware-system-csi
+rules:
+  - apiGroups: [""]
+    resources: ["configmaps"]
+    verbs: ["get", "list", "watch"]
+---
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: vsphere-csi-webhook-role-binding
+  namespace: vmware-system-csi
+subjects:
+  - kind: ServiceAccount
+    name: default
+    namespace: vmware-system-csi
+roleRef:
+  kind: Role
+  name: vsphere-csi-webhook-role
+  apiGroup: rbac.authorization.k8s.io
+---
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: vsphere-csi-webhook
+  namespace: vmware-system-csi
+  labels:
+    app: vsphere-csi-webhook
+spec:
+  replicas: 3
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 1
+      maxSurge: 0
+  selector:
+    matchLabels:
+      app: vsphere-csi-webhook
+  template:
+    metadata:
+      labels:
+        app: vsphere-csi-webhook
+    spec:
+      affinity:
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            - labelSelector:
+                matchExpressions:
+                  - key: app
+                    operator: In
+                    values:
+                      - vsphere-csi-webhook
+              topologyKey: kubernetes.io/hostname
+      hostNetwork: true
```

Comment:
why do we need to set hostNetwork: true?

Reply:
Will check if webhook works fine without the configuration

Reply:
Removing `hostNetwork: true` will end up with the following issue while deploying webhook pod.

```
86s         Warning   FailedCreatePodSandBox   pod/vsphere-csi-webhook-77b88dbc88-k6dl8    Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "756db6a6233f4df60e55f19c25026a61f6a74cd607d08b51b84a05214c0c66fa": failed to find plugin "nsx" in path [/opt/cni/bin]
73s         Warning   FailedCreatePodSandBox   pod/vsphere-csi-webhook-77b88dbc88-k6dl8    Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "b5fc44fea15e460cb2d77317b417b6f4a00b686745a4173b99405982e9091199": failed to find plugin "nsx" in path [/opt/cni/bin]
61s         Warning   FailedCreatePodSandBox   pod/vsphere-csi-webhook-77b88dbc88-k6dl8    Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "3d5dce327fb6c016e41758d4fd68d275495abd18520c50b13cc8cb38a5ebc0b4": failed to find plugin "nsx" in path [/opt/cni/bin]
49s         Warning   FailedCreatePodSandBox   pod/vsphere-csi-webhook-77b88dbc88-k6dl8    Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "a4c85638d82e34ec0a6d251e9611c86585fae1ceeaaddffdd8f09c65c1389d37": failed to find plugin "nsx" in path [/opt/cni/bin]
34s         Warning   FailedCreatePodSandBox   pod/vsphere-csi-webhook-77b88dbc88-k6dl8    Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "c635270f301bfa36610d7e43bb45a174e5f808dae7c817d91b38c11ffcf725b8": failed to find plugin "nsx" in path [/opt/cni/bin]
20s         Warning   FailedCreatePodSandBox   pod/vsphere-csi-webhook-77b88dbc88-k6dl8    Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "6a7a00ad14379fe4b818922a09c60a658c815822f5db5da15e46d11ddf4aa1b6": failed to find plugin "nsx" in path [/opt/cni/bin]
7s          Warning   FailedCreatePodSandBox   pod/vsphere-csi-webhook-77b88dbc88-k6dl8    Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "5a15c3c2966331c1cb11513cdd6e149f117d0d8edf5a7f86cc586287b6476795": failed to find plugin "nsx" in path [/opt/cni/bin]
```

So, I will just keep it.